### PR TITLE
[Ascend] fuj/aclnn-replace-dropout

### PR DESCRIPTION
**Pull request by jingguo-st**
## Motivation and Context

1. replace diopiDropout and diopiDropoutInp
2. impl makeZerosLike

## Checklist
**Before PR**:

- [ ] I have read and followed the workflow indicated in the [Contributors.md](https://github.com/DeepLink-org/DIOPI/blob/main/Contributors.md) to create this PR.
- [ ] Pre-commit or linting tools indicated in [Contributors.md](https://github.com/DeepLink-org/DIOPI/blob/main/Contributors.md) are used to fix the potential lint issues.
- [ ] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] New functionalities are covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [ ] The documentation has been modified accordingly, including docstring or example tutorials.

**After PR**:

- [ ] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

File: impl/ascend/common/utils.cpp [hunk 1]
```diff
@@ -372,6 +372,18 @@ diopiError_t makeOnesLike(diopiContextHandle_t ctx, diopiTensorHandle_t* out, di
     return makeOnesLike(ctx, out, src, dtype);
 }
 
+diopiError_t makeZerosLike(diopiContextHandle_t ctx, diopiTensorHandle_t* out, diopiConstTensorHandle_t src, diopiDtype_t dtype) {
+    makeTensorLike(ctx, out, src, dtype);
+    fillTensor(ctx, *out, static_cast<float>(0.0));
```
Comment (yangbofun) [inline impl/ascend/common/utils.cpp:377]: 有aclnninplaceone，直接调用这个。

File: impl/ascend/functions/dropout.cpp [hunk 2]
```diff
@@ -4,84 +4,116 @@
...
+    ASCEND_CHECK_ABORT(ret == diopiSuccess, "[npuDropout2dOut] require tensor for mask failed.");
+
+    uint64_t seed, offset;
+    DIOPI_CALL(diopiGeneratorGetSeedAndOffset(generator, &seed, &offset));
+
+    DIOPI_ASCEND_CALL_ACLNN(aclnnDropoutGenMask, ctx, inAt.shape(), p, seed, offset, maskNpu);
+    DIOPI_ASCEND_CALL_ACLNN(aclnnDropoutDoMask, ctx, input2d, maskNpu, p, out2d);
+    DIOPI_ASCEND_CALL_ACLNN(aclnnMul, ctx, input, out2d, out);
+
+    auto ref = constructDiopiScalarT(diopi_dtype_float32, 0.0);
```
Comment (yangbofun) [inline impl/ascend/functions/dropout.cpp:61]: 这里写死float32的话，如果上游传来的input是fp16或bf16，可能会导致性能问题。看下华为aclnnNeScalar接口关于类型的说明。

File: impl/ascend/common/utils.cpp (hunk 1 above)
Comment (jingguo-st) [reply impl/ascend/common/utils.cpp]: fix

File: impl/ascend/functions/dropout.cpp (hunk 2 above)
Comment (jingguo-st) [reply impl/ascend/functions/dropout.cpp]: fix

**Review by zhangzefeng92 — APPROVED** [review]

**Review by POI-WX — APPROVED** [review]

**Review by yangbofun — APPROVED** [review]

**Pull request merged**